celements-model/src/main/java/com/celements/model/context/DefaultModelContext.java

```diff
@@ -260,6 +271,26 @@ private String getDefaultLangFromConfigSrc(EntityReference ref) {
     }
   }
 
+  @Override
+  public XWikiDocument getOrCreateXWikiPreferenceDoc() {
+    DocumentReference docRef = new RefBuilder().space(XWIKI_SPACE).doc(XWIKI_PREF_DOC_NAME).build(
+        DocumentReference.class);
+    return getModelAccess().getOrCreateDocument(docRef);
+  }
+
+  @Override
+  public XWikiDocument getOrCreateSpacePreferenceDoc() {
```

Comment:
Please split in two methods:
1. XWikiDocument getOrCreateSpacePreferenceDoc(SpaceReference spaceRef)
2. SpaceReference getCurrentSpaceReference()
this allows general use and better reuse of both methods instead of a combined one.

Reply:
parameter `SpaceReference spaceRef` is still missing. The current-space is a special case. I think we do not need to handle this special case in its own method. Instead make the `getOrCreateSpacePreferenceDoc` more general and chain the `getCurrentSpaceReference` with the `getOrCreateSpacePreferenceDoc(getCurrentSpaceReference())` in the caller.